ci/scripts/java_jni_manylinux_build.sh

```diff
@@ -115,7 +115,7 @@ if [ "${ARROW_BUILD_TESTS}" = "ON" ]; then
       # ImportError: ../grpc/_cython/cygrpc.cpython-38-aarch64-linux-gnu.so:
       # undefined symbol: vtable for std::__cxx11::basic_ostringstream<
       #   char, std::char_traits<char>, std::allocator<char> >
-      exclude_tests="${exclude_tests}|arrow-gcsfs-test"
+      exclude_tests="${exclude_tests}|arrow-gcsfs-test|arrow-dataset-dataset-writer-test"
```

Comment:
nit: can you link the GitHub issue explaining why we are skipping this test?

Reply:
It is a temporary workaround that only applies to testing this PR's scope. I think we should wait until https://github.com/apache/arrow/issues/39326 is resolved. Please let me know if this sound great for you, or, do we should push this PR with this exclude test to main branch?

Reply:
Are we sure #39326 will be resolved in time?